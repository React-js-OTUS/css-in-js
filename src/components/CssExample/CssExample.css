.CssExample__button {
  border-radius: 8px;
  font-size: 18px;
  padding: 8px 16px;
  outline: none;
  border: 0;
  background-color: #00CC00;
  color: #fff;
  cursor: pointer;
}

.CssExample__button:hover {
  background-color: #39E639;
}

.CssExample__button:active {
  background-color: #269926;
}

.CssExample__text {
  font-size: 18px;
}

.CssExample__text_opened {
  display: block;
}

.CssExample__text_closed {
  display: none;
}
