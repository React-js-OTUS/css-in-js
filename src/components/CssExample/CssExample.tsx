import React, { FC, useState } from 'react';
import cn from 'clsx';
import './CssExample.css';

export type ButtonProps = {
  className?: string;
};

export const CssExample: FC<ButtonProps> = ({ className, ...props }) => {
  const [opened, setOpened] = useState<boolean>(false);
  return (
    <div className={cn('css-example', className)}>
      <button className={'css-example__button'} {...props} onClick={() => setOpened(v => !v)} />
      <div className={cn('css-example__text', opened ? 'css-example__text_opened' : 'css-example__text_closed')}>Test</div>
      <div className={cn('css-example__text', !opened && 'css-example__text_closed')}>Test</div>
    </div>
  );
}
