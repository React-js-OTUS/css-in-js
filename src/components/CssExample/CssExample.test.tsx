import React from 'react';
import { render } from '@testing-library/react';
import { CssExample } from './CssExample';

describe('CssButton', () => {
  it('base snapshot', () => {
    const { container } = render(<CssExample>CssButton</CssExample>);
    expect(container).toMatchSnapshot();
  });
});
