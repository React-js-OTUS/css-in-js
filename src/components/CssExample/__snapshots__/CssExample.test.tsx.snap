// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CssButton base snapshot 1`] = `
<div>
  <div
    class="css-example"
  >
    <button
      class="css-example__button"
    >
      CssButton
    </button>
    <div
      class="css-example__text css-example__text_closed"
    >
      Test
    </div>
    <div
      class="css-example__text css-example__text_closed"
    >
      Test
    </div>
  </div>
</div>
`;
